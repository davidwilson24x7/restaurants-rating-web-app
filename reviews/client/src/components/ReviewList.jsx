import React from 'react';
import ReactDOM from 'react-dom';
import ReviewItem from './ReviewItem.jsx';
import Pagination from './Pagination.jsx';

import styled from 'styled-components';

// need a write review item
// pass down user's profile
// hover -> star
// make a modal that lets you write a review and submit

const UnorderedList = styled.ul `
  list-style: none;
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
  padding-inline-start: 40px;
  color: #333;
`

const ReviewList = (props) => {
  const numPages = Math.ceil(props.allReviews.length/3);
  let pagesArr = [];
  for (let i = 1; i < numPages + 1; i++) {
    pagesArr.push(i)
  }

  return (
    <div>
      <UnorderedList>
        {
          props.allReviews.map((item) => {
            return <li><ReviewItem reviewItem={item} buttonSubmit={props.buttonSubmit}/></li>
          })
        }
      </UnorderedList>

      {/* <span>
        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" class="icon_svg"><path d="M14.25 17.58a1 1 0 0 1-.71-.3L9 12.7a1 1 0 0 1 0-1.4l4.5-4.58A1 1 0 0 1 15 6.7a1 1 0 0 1 0 1.42L11.15 12 15 15.88a1 1 0 0 1 0 1.42 1 1 0 0 1-.75.28z"></path></svg>
      {
        pagesArr.map((pageNum) => {
          return <Pagination pageNum={pageNum}/>
        })
      }
        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" class="icon_svg"><path d="M9.75 17.58a1 1 0 0 1-.7-.28 1 1 0 0 1 0-1.42l3.8-3.88L9 8.12a1 1 0 1 1 1.41-1.42L15 11.3a1 1 0 0 1 0 1.4l-4.5 4.58a1 1 0 0 1-.75.3z"></path></svg>
      </span> */}
    </div>
)}

export default ReviewList;
