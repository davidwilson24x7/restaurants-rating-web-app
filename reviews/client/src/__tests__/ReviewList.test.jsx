import styled from 'styled-components';
import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import ReviewList from '../components/ReviewList';


describe('ReviewList Test Suite', () => {
  const reviewList = [
      {
        "id": 2,
        "stars": 2,
        "uploadDate": "3/23/2020",
        "restaurantVisit": null,
        "content": "LOVE THIS PLACE",
        "emojiUseful": 1,
        "emojiFunny": 1,
        "emojiCool": null,
        "reply": null,
        "replyDate": null,
        "userName": "Minji Gwakster",
        "restaurantName": "Noodle Soupz",
        "pictures": []
    },
    {
        "id": 3,
        "stars": 4,
        "uploadDate": "2/12/2020",
        "restaurantVisit": 2,
        "content": "Chicken is finger licking gooood",
        "emojiUseful": 1,
        "emojiFunny": null,
        "emojiCool": 1,
        "reply": "Appreciate it!",
        "replyDate": "2/13/2020",
        "userName": "Billy Wonka",
        "restaurantName": "Kyochon Chicken",
        "pictures": ["https://pic4.jpg", "https://pic5.jpg"]
    }
  ]

  it('should render reviews', () => {
    const component = <ReviewList allReviews={reviewList}/>

    const list = renderer.create(component).toJSON()
    expect(list).toMatchSnapshot()
  })




})