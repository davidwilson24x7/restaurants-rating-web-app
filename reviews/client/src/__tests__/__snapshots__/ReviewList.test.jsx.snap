// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList Test Suite should render reviews 1`] = `
<ul
  className="sc-fzoyAV jBeYyC"
>
  <li
    className="sc-fzoLag geROkq"
  >
    <div>
      <div>
        <span>
          2
           Stars
        </span>
        <span
          className="sc-AxhUy gWmJES"
        >
          3/23/2020
        </span>
      </div>
      <div />
      <div
        className="sc-Axmtr ePJevX"
      >
        LOVE THIS PLACE
      </div>
      <div>
        <span
          className="sc-fzoLsD jdrKxG"
        >
          <svg
            class="icon_svg"
            className="sc-fzozJi bZJaSO"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 1a11 11 0 0 1 11 11c0 6.075-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1zm0 20a9 9 0 1 0 0-18 9 9 0 0 0 0 18zm0-15.82a5.56 5.56 0 0 1 3 10.26V17a2 2 0 0 1-2 2h-2a2 2 0 0 1-2-2v-1.56a5.56 5.56 0 0 1 3-10.26zM11 17h2v-1h-2v1zm2.56-3a3.58 3.58 0 1 0-3.12 0h3.12z"
            />
          </svg>
           Useful 
          1
        </span>
        <span
          className="sc-fzoLsD jdrKxG"
        >
          <svg
            class="icon_svg"
            className="sc-fzozJi bZJaSO"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 23C5.925 23 1 18.075 1 12S5.925 1 12 1s11 4.925 11 11a11 11 0 0 1-11 11zm0-20a9 9 0 1 0 0 18 9 9 0 0 0 0-18zm-4.5 8a1.5 1.5 0 0 1 3 0h-3zm6 0a1.5 1.5 0 0 1 3 0h-3zm-7.21 2h11.46a.26.26 0 0 1 .25.29c-.57 3.25-3 5.71-6 5.71s-5.43-2.46-5.96-5.71a.26.26 0 0 1 .25-.29z"
            />
          </svg>
           Funny 
          1
        </span>
        <span
          className="sc-fzoLsD jdrKxG"
        >
          <svg
            class="icon_svg"
            className="sc-fzozJi bZJaSO"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 23C5.925 23 1 18.075 1 12S5.925 1 12 1s11 4.925 11 11a11 11 0 0 1-11 11zm0-20a9 9 0 1 0 0 18 9 9 0 0 0 0-18zm-7.48 7.56a.43.43 0 0 1 .41-.56h14.14a.43.43 0 0 1 .41.56l-.85 2.53a3 3 0 0 1-2.8 2 3 3 0 0 1-2.8-2l-.28-.85a.8.8 0 0 0-.75-.54.8.8 0 0 0-.75.54l-.28.85a3 3 0 0 1-2.8 2 3 3 0 0 1-2.8-2l-.85-2.53zm3.18 5.63a16.3 16.3 0 0 0 8.6 0 .25.25 0 0 1 .26.39A5.71 5.71 0 0 1 12 19a5.71 5.71 0 0 1-4.56-2.42.25.25 0 0 1 .26-.39z"
            />
          </svg>
          Cool 
        </span>
        <div
          className="sc-fzpans gVIgpt"
        >
          <span
            className="sc-fzplWN SVtFF"
          >
            <svg
              class="icon_svg"
              className="sc-fznyAO jdhGMs"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 10V3c4.976 1.098 4.024-1 8 0v7c-4.024-.976-3.024 1.024-8 0zM4 2h1v14H4V2z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </li>
  <li
    className="sc-fzoLag geROkq"
  >
    <div>
      <div>
        <span>
          4
           Stars
        </span>
        <span
          className="sc-AxhUy gWmJES"
        >
          2/12/2020
        </span>
      </div>
      <div>
        <span
          className="sc-AxiKw iYHdRM"
        >
          <svg
            class="icon_svg"
            className="sc-AxhCb jMlIQM"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/CheckinIcon"
          >
            <path
              d="M15 15H3a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2h2a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2zM9 5a4 4 0 1 0 0 8 4 4 0 0 0 0-8zm0 6.5a2.5 2.5 0 1 1 0-5 2.5 2.5 0 0 1 0 5z"
            />
          </svg>
           
          2
           photos
        </span>
        <span
          className="sc-AxgMl hSRzca"
        >
          <svg
            class="icon_svg"
            className="sc-AxheI hwhcgj"
            fill="#0077bc"
            height="18px"
            viewBox="0 0 18 18"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 9l-2.136-1.84.932-2.66-2.772-.525-.524-2.77-2.66.93L8.997 0 7.163 2.136 4.5 1.206l-.525 2.77-2.77.524.932 2.66L0 9l2.137 1.84-.932 2.66 2.77.525.526 2.77 2.664-.932L8.998 18l1.84-2.137 2.662.932.524-2.77 2.772-.524-.932-2.66L18 9zm-9.85 3.23L5.324 9.4l1.13-1.13 1.698 1.696 3.396-3.395 1.13 1.134-4.525 4.525z"
            />
          </svg>
          2
           check-in
        </span>
      </div>
      <div
        className="sc-Axmtr ePJevX"
      >
        Chicken is finger licking gooood
      </div>
      <span
        className="sc-AxjAm eqkatP"
      >
        <img
          className="sc-AxirZ bsiIMd"
          src="https://pic4.jpg"
        />
      </span>
      <span
        className="sc-AxjAm eqkatP"
      >
        <img
          className="sc-AxirZ bsiIMd"
          src="https://pic5.jpg"
        />
      </span>
      <div
        className="sc-AxmLO eFtarN"
      >
        See all photos from 
        Billy Wonka
         for 
        Kyochon Chicken
      </div>
      <div>
        <span
          className="sc-fzoLsD jdrKxG"
        >
          <svg
            class="icon_svg"
            className="sc-fzozJi bZJaSO"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 1a11 11 0 0 1 11 11c0 6.075-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1zm0 20a9 9 0 1 0 0-18 9 9 0 0 0 0 18zm0-15.82a5.56 5.56 0 0 1 3 10.26V17a2 2 0 0 1-2 2h-2a2 2 0 0 1-2-2v-1.56a5.56 5.56 0 0 1 3-10.26zM11 17h2v-1h-2v1zm2.56-3a3.58 3.58 0 1 0-3.12 0h3.12z"
            />
          </svg>
           Useful 
          1
        </span>
        <span
          className="sc-fzoLsD jdrKxG"
        >
          <svg
            class="icon_svg"
            className="sc-fzozJi bZJaSO"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 23C5.925 23 1 18.075 1 12S5.925 1 12 1s11 4.925 11 11a11 11 0 0 1-11 11zm0-20a9 9 0 1 0 0 18 9 9 0 0 0 0-18zm-4.5 8a1.5 1.5 0 0 1 3 0h-3zm6 0a1.5 1.5 0 0 1 3 0h-3zm-7.21 2h11.46a.26.26 0 0 1 .25.29c-.57 3.25-3 5.71-6 5.71s-5.43-2.46-5.96-5.71a.26.26 0 0 1 .25-.29z"
            />
          </svg>
           Funny 
        </span>
        <span
          className="sc-fzoLsD jdrKxG"
        >
          <svg
            class="icon_svg"
            className="sc-fzozJi bZJaSO"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 23C5.925 23 1 18.075 1 12S5.925 1 12 1s11 4.925 11 11a11 11 0 0 1-11 11zm0-20a9 9 0 1 0 0 18 9 9 0 0 0 0-18zm-7.48 7.56a.43.43 0 0 1 .41-.56h14.14a.43.43 0 0 1 .41.56l-.85 2.53a3 3 0 0 1-2.8 2 3 3 0 0 1-2.8-2l-.28-.85a.8.8 0 0 0-.75-.54.8.8 0 0 0-.75.54l-.28.85a3 3 0 0 1-2.8 2 3 3 0 0 1-2.8-2l-.85-2.53zm3.18 5.63a16.3 16.3 0 0 0 8.6 0 .25.25 0 0 1 .26.39A5.71 5.71 0 0 1 12 19a5.71 5.71 0 0 1-4.56-2.42.25.25 0 0 1 .26-.39z"
            />
          </svg>
          Cool 
          1
        </span>
        <div
          className="sc-fzpans gVIgpt"
        >
          <span
            className="sc-fzplWN SVtFF"
          >
            <svg
              class="icon_svg"
              className="sc-fznyAO jdhGMs"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 10V3c4.976 1.098 4.024-1 8 0v7c-4.024-.976-3.024 1.024-8 0zM4 2h1v14H4V2z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="sc-fznKkj cgFSfQ"
      >
        <span
          className="sc-fznZeY cuWJrp"
        >
          Comment from 
          Kyochon Chicken
        </span>
        <br />
        <span
          className="sc-fzqNJr jemXeG"
        >
          2/13/2020
          <span
            className="sc-fzqBZW dpateo"
          >
            â€¢
          </span>
          Appreciate it!
        </span>
      </div>
    </div>
  </li>
</ul>
`;
